server/memphis_handler_schemas.go

```diff
@@ -221,104 +169,46 @@ func (sh SchemasHandler) CreateNewSchema(c *gin.Context) {
 		return
 	}
-	filter := bson.M{"name": schemaName}
-	var schema models.Schema
-	err = schemasCollection.FindOne(context.TODO(), filter).Decode(&schema)
-	if err != nil && !strings.Contains(err.Error(), "no documents") {
-		serv.Errorf("CreateSchema error: " + err.Error())
-		c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
-		return
-	}
-	versionNumber := len(schema.Versions) + 1
-	var active bool
-	if versionNumber == 1 {
-		active = true
-	} else {
-		active = false
-	}
 	newSchemaVersion := models.SchemaVersion{
 		ID:            primitive.NewObjectID(),
-		VersionNumber: versionNumber,
-		Active:        active,
+		VersionNumber: 1,
+		Active:        true,
 		CreatedByUser: user.Username,
 		CreationDate:  time.Now(),
 		SchemaContent: schemaContent,
 	}
 	newSchema := models.Schema{
-		ID:        primitive.NewObjectID(),
-		Name:      schemaName,
-		Type:      schemaType,
-		Versions:  schema.Versions,
-		IsDeleted: false,
+		ID:       primitive.NewObjectID(),
+		Name:     schemaName,
+		Type:     schemaType,
+		Versions: []primitive.ObjectID{newSchemaVersion.ID},
 	}
 
-	filter = bson.M{"name": newSchema.Name, "is_deleted": false}
-	update := bson.M{
-		"$setOnInsert": bson.M{
-			"_id":  newSchema.ID,
-			"name": newSchema.Name,
-			"type": newSchema.Type,
-		},
-		"$push": bson.M{"versions": newSchemaVersion.ID},
-	}
-	opts := options.Update().SetUpsert(true)
-	updateResults, err := schemasCollection.UpdateOne(context.TODO(), filter, update, opts)
+	_, err = schemasCollection.InsertOne(context.TODO(), newSchema)
 	if err != nil {
 		serv.Errorf("CreateSchema error: " + err.Error())
 		c.AbortWithStatusJSON(500, gin.H{"message": "Server error"})
 		return
 	}
-	if updateResults.MatchedCount > 0 {
-		serv.Warnf("Scheam with the same name is already exist")
-		newSchema.ID = schema.ID
-	} else {
-		message := "Schema " + schemaName + " has been created"
-		serv.Noticef(message)
-		var auditLogs []interface{}
-		newAuditLog := models.AuditLog{
-			ID:            primitive.NewObjectID(),
-			StationName:   schemaName,
-			Message:       message,
-			CreatedByUser: user.Username,
-			CreationDate:  time.Now(),
-			UserType:      user.UserType,
-		}
-		auditLogs = append(auditLogs, newAuditLog)
-		err = CreateAuditLogs(auditLogs)
-		if err != nil {
-			serv.Warnf("CreateSchema error: " + err.Error())
-		}
-	}
-
-	filter = bson.M{"_id": newSchemaVersion.ID, "is_deleted": false}
-	update = bson.M{
-		"$setOnInsert": bson.M{
-			"_id":             newSchemaVersion.ID,
-			"version_number":  newSchemaVersion.VersionNumber,
-			"active":          newSchemaVersion.Active,
-			"created_by_user": newSchemaVersion.CreatedByUser,
-			"creation_date":   newSchemaVersion.CreationDate,
-			"schema_content":  newSchemaVersion.SchemaContent,
-		},
-	}
-	_, err = schemasVersionCollection.UpdateOne(context.TODO(), filter, update, opts)
 
+	_, err = schemaVersionCollection.InsertOne(context.TODO(), newSchemaVersion)
```

Comment:
this should be insertOne as you did but put this one + the log of newly created schema in case the matchCount == 0, otherwise return already exist error to the user